src/main/java/ro/unibuc/hello/data/Account.java

```diff
@@ -0,0 +1,65 @@
+package ro.unibuc.hello.data;
+
+import com.mongodb.lang.NonNull;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.Document;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Set;
+
+@Document(collection = "accounts")
+public class Account{
+
+
+    @Id @NonNull private String email;
+    @NonNull private String firstName;
+    @NonNull private String lastName;
+    @NonNull private String password;
+    @NonNull private Date dateCreated;
+    private Set<Role> roles = new HashSet<>();
+    public Account(@NotNull String email, @NotNull String firstName, @NotNull String lastName, String password) {
+        this.email = email;
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.dateCreated = new Date();
+        this.password = new BCryptPasswordEncoder().encode(password);
```

Comment:
1. this is not good practice to add functionality in constructors
2. and the reason why the /login POST is falling - the Account object is used in the custom implementation of userDetailsService in CustomUserDetailsService.loadUserByUsername(email) => every time an Account get instantiated the password gets encoded, making the login check to return bad credentials

Reply:
you could change this `this.password = new BCryptPasswordEncoder().encode(password);` to `this.password = password;` and add the encoding in the part where you add the two users in the db (HelloApplication.runAfterObjectCreated() `Account radu = new Account("radu.ndlcu@gmail.com",
				"Radu", "Nedelcu", passwordEncoder.encode("1234"))`)

Reply:
Thanks for the feedback, managed to fix it.